lms/djangoapps/instructor/views/api.py

```diff
@@ -2625,24 +2628,77 @@ def enable_certificate_generation(request, course_id=None):
     return redirect(_instructor_dash_url(course_key, section='certificates'))
 
 
-#---- Gradebook (shown to small courses only) ----
-@cache_control(no_cache=True, no_store=True, must_revalidate=True)
-@require_level('staff')
-def spoc_gradebook(request, course_id):
+def calculate_page_info(offset):
     """
-    Show the gradebook for this course:
-    - Only shown for courses with enrollment < settings.FEATURES.get("MAX_ENROLLMENT_INSTR_BUTTONS")
-    - Only displayed to course staff
+    This method calculates query offsets for next and previous page, current page number on give
+    current query offset.
+
+    :param offset: offset for database query
+    :return: tuple consist of page number, query offset for next and previous pages and valid offset
     """
-    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
-    course = get_course_with_access(request.user, 'staff', course_key, depth=None)
 
-    enrolled_students = User.objects.filter(
+    # validate offset
+    if not isinstance(offset, int) and not offset.isdigit():
+        offset = 0
+    elif offset < 0:
+        offset = 0
```

Comment:
let me refactor it
On 04-Nov-2015 11:52 pm, "Giovanni Di Milia" <notifications@github.com>
wrote:

> In lms/djangoapps/instructor/views/api.py
> <https://github.com/edx/edx-platform/pull/10250#discussion_r43921497>:
>
> >
> > -    enrolled_students = User.objects.filter(
> > +    # validate offset
> > +    if not isinstance(offset, int) and not offset.isdigit():
> > +        offset = 0
> > +    elif offset < 0:
> > +        offset = 0
>
> I think these two lines can be refactored as:
>
> if not (isinstance(offset, int) or offset.isdigit()) or offset < 0:
>     offset = 0
>
> Python evaluates conditions from left to right
> <https://docs.python.org/2/library/stdtypes.html#boolean-operations-and-or-not>
> so it will not get to the offset < 0 unless the first condition has not
> been matched.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/10250/files#r43921497>.
>